apps/music/js/ui/views/player_view.js

```diff
@@ -1095,7 +1095,8 @@ var PlayerView = {
         // If we're within 1 second of the end of the song, register
         // a timeout to skip to the next song one second after the song ends
         if (this.audio.currentTime >= this.audio.duration - 1 &&
-            this.endedTimer == null) {
+            this.endedTimer == null &&
+            this.playStatus !== PLAYSTATUS_PAUSED) {
           var timeToNext = (this.audio.duration - this.audio.currentTime + 1);
           this.endedTimer = setTimeout(function() {
```

Comment:
I wonder if we even need the `setTimeout` anymore. I seem to recall this being a workaround for the `'ended'` event not firing for MP3s, but maybe that's been fixed in Gecko. If so, we might be able to remove all of this code.